src/Optimizer/Configuration/MinifyHtmlConfiguration.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace AmpProject\Optimizer\Configuration;
+
+use AmpProject\Optimizer\Exception\InvalidConfigurationValue;
+
+/**
+ * Configuration for the MinifyHtml transformer.
+ *
+ * @property bool $minify                Is minification enabled.
+ * @property bool $minifyAmpScript       Is amp-script minification enabled.
+ * @property bool $minifyJSON            Is JSON data minification enabled.
+ * @property bool $collapseWhitespace    Is collapsing the whitespace enabled.
+ * @property bool $removeComments        Should remove comments.
+ * @property bool $canCollapseWhitespace Can collapse whitespace
+ * @property bool $inBody                Is node in the body
```

Comment:
As noted further down, the `$canCollapseWhitespace` and `$inBody` values are not actually configuration values, but rather node iteration state. They need to be removed here and added as method arguments in the transformer.